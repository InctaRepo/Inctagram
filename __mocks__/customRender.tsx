/* eslint-disable */
import React, { ReactElement } from 'react'
import { Provider } from 'react-redux'

import { store } from '@/store'
import { render, renderHook, RenderOptions } from '@testing-library/react'

export const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    // <ThemeProvider theme="light">
    //   <TranslationProvider messages={defaultStrings}>
    <Provider store={store}>{children}</Provider>
    //   </TranslationProvider>
    // </ThemeProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) => {
  return renderHook(() => hook(), { wrapper: AllTheProviders, ...options })
}
export * from '@testing-library/react'
export * from '@testing-library/user-event'
// eslint-disable-next-line import/export
export { customRender as render }
