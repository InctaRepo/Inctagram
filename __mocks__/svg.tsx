import React, { SVGProps, forwardRef } from 'react'

const SvgrMock = forwardRef<SVGSVGElement, SVGProps<SVGSVGElement>>((props, ref) => (
  <svg ref={ref} {...props} />
))

export const ReactComponent = SvgrMock
export default SvgrMock
