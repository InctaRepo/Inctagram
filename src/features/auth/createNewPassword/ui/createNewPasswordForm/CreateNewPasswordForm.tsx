import React, { useEffect } from 'react'

import { DevTool } from '@hookform/devtools'
import { zodResolver } from '@hookform/resolvers/zod'
import { useForm } from 'react-hook-form'

import s from './createNewPasswordForm.module.scss'

import { FormFields, triggerZodFieldError } from '@/src/shared/helpers/updateZodError'
import { useTranslate } from '@/src/shared/hooks'
import { PasswordsMatchForm, passwordsMatchSchema } from '@/src/shared/schemas/passwordsMatchSchema'
import { Button } from '@/src/shared/ui/button'
import { ControlledTextField } from '@/src/shared/ui/controlled'
import { Typography } from '@/src/shared/ui/typography'

type CreateNewPasswordProps = {
  onSubmitHandler: (data: PasswordsMatchForm) => void
}

export const CreateNewPasswordForm = ({ onSubmitHandler }: CreateNewPasswordProps) => {
  const { t } = useTranslate()

  const {
    control,
    handleSubmit,
    formState,
    trigger,
    formState: { touchedFields, errors },
  } = useForm<PasswordsMatchForm>({
    resolver: zodResolver(passwordsMatchSchema(t)),
    mode: 'onTouched',
    defaultValues: {
      password: '',
      passwordConfirm: '',
    },
  })

  useEffect(() => {
    const touchedFieldNames: FormFields[] = Object.keys(touchedFields) as FormFields[]

    triggerZodFieldError(touchedFieldNames, trigger)
  }, [t])

  const onSubmit = handleSubmit((data: PasswordsMatchForm) => {
    onSubmitHandler(data)
  })

  return (
    <form className={s.wrapper} onSubmit={onSubmit}>
      <Typography variant={'h1'}>{t.auth.createNewPassword}</Typography>
      <DevTool control={control} />

      <ControlledTextField
        control={control}
        name={'password'}
        type={'password'}
        label={t.auth.newPassword}
        className={s.password}
      />

      <ControlledTextField
        control={control}
        name={'passwordConfirm'}
        type={'password'}
        label={t.auth.passwordConfirmation}
        className={`${s.password} ${errors.passwordConfirm && s.fieldWithError}`}
      />
      <div className={s.text}>
        <Typography variant="medium14" className={s.passwordRequirement}>
          {t.auth.passwordCharacters}
        </Typography>
      </div>
      <Button type={'submit'} variant="primary" fullWidth={true} className={s.btn}>
        <Typography variant="bold16">{t.auth.createNewPassword}</Typography>
      </Button>
    </form>
  )
}
