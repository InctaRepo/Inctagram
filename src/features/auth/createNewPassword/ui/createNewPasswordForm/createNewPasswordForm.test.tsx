import React from 'react'

import { render, screen, userEvent, waitFor } from '@/__mocks__/customRender'

import { CreateNewPasswordForm } from './CreateNewPasswordForm'

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({ locale: 'en' }),
}))

function setup(jsx: React.JSX.Element) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  }
}

const onSubmitHandler = jest.fn()

describe('CreateNewPasswordForm', () => {
  it('renders form fields and submits data', async () => {
    const onSubmitHandler = jest.fn()
    const { container, user } = setup(<CreateNewPasswordForm onSubmitHandler={onSubmitHandler} />)

    expect(container.querySelector('form')).toBeInTheDocument()
    await user.type(screen.getByRole('password', { name: /password/i }), '1qaz@WSX')
    await user.type(screen.getByRole('passwordConfirm', { name: /password/i }), '1qaz@WSX')
    await user.click(screen.getByRole('button', { name: /Create new password/i }))
    expect(onSubmitHandler).toHaveBeenCalledTimes(1)
    expect(onSubmitHandler).toHaveBeenCalledWith({
      password: '1qaz@WSX',
      passwordConfirm: '1qaz@WSX',
    })
  })
  it('snapshot CreateNewPasswordForm', () => {
    const snapshot = setup(<CreateNewPasswordForm onSubmitHandler={onSubmitHandler} />)

    waitFor(() => expect(snapshot).toMatchSnapshot())
  })
})
