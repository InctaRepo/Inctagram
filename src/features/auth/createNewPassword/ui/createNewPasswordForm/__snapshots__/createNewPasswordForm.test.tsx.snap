// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewPasswordForm snapshot CreateNewPasswordForm 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="wrapper"
      >
        <p
          class="h1 primary  undefined"
        >
          Create new password
        </p>
        <div
          class="password "
        >
          <div
            class="textFieldWrap"
          >
            <label
              class="label undefined"
              data-testid="password"
              for="password"
            >
              New password
            </label>
            <input
              autocomplete="new-password"
              class="textField undefined undefined"
              id="password"
              name="password"
              role="password"
              type="password"
              value=""
            />
            <svg
              class="iconEye undefined"
            />
          </div>
        </div>
        <div
          class="password undefined "
        >
          <div
            class="textFieldWrap"
          >
            <label
              class="label undefined"
              data-testid="passwordConfirm"
              for="passwordConfirm"
            >
              Password confirmation
            </label>
            <input
              autocomplete="new-password"
              class="textField undefined undefined"
              id="passwordConfirm"
              name="passwordConfirm"
              role="passwordConfirm"
              type="password"
              value=""
            />
            <svg
              class="iconEye undefined"
            />
          </div>
        </div>
        <div
          class="text"
        >
          <p
            class="medium14 primary  passwordRequirement"
          >
            Your password must be between 6 and 20 characters
          </p>
        </div>
        <button
          class="primary fullWidth btn"
          type="submit"
        >
          <p
            class="bold16 primary  undefined"
          >
            Create new password
          </p>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="wrapper"
    >
      <p
        class="h1 primary  undefined"
      >
        Create new password
      </p>
      <div
        class="password "
      >
        <div
          class="textFieldWrap"
        >
          <label
            class="label undefined"
            data-testid="password"
            for="password"
          >
            New password
          </label>
          <input
            autocomplete="new-password"
            class="textField undefined undefined"
            id="password"
            name="password"
            role="password"
            type="password"
            value=""
          />
          <svg
            class="iconEye undefined"
          />
        </div>
      </div>
      <div
        class="password undefined "
      >
        <div
          class="textFieldWrap"
        >
          <label
            class="label undefined"
            data-testid="passwordConfirm"
            for="passwordConfirm"
          >
            Password confirmation
          </label>
          <input
            autocomplete="new-password"
            class="textField undefined undefined"
            id="passwordConfirm"
            name="passwordConfirm"
            role="passwordConfirm"
            type="password"
            value=""
          />
          <svg
            class="iconEye undefined"
          />
        </div>
      </div>
      <div
        class="text"
      >
        <p
          class="medium14 primary  passwordRequirement"
        >
          Your password must be between 6 and 20 characters
        </p>
      </div>
      <button
        class="primary fullWidth btn"
        type="submit"
      >
        <p
          class="bold16 primary  undefined"
        >
          Create new password
        </p>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
